store/gaskvstore.go

```diff
@@ -0,0 +1,148 @@
+package store
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// nolint
+const (
+	HasCost          = 10
+	ReadCostFlat     = 10
+	ReadCostPerByte  = 1
+	WriteCostFlat    = 10
+	WriteCostPerByte = 10
+	KeyCostFlat      = 5
+	ValueCostFlat    = 10
+	ValueCostPerByte = 1
+)
+
+// gasKVStore applies gas tracking to an underlying kvstore
+type gasKVStore struct {
+	gasMeter sdk.GasMeter
+	parent   sdk.KVStore
+}
+
+// nolint
+func NewGasKVStore(gasMeter sdk.GasMeter, parent sdk.KVStore) *gasKVStore {
+	kvs := &gasKVStore{
+		gasMeter: gasMeter,
+		parent:   parent,
+	}
+	return kvs
+}
+
+// Implements Store.
+func (gi *gasKVStore) GetStoreType() sdk.StoreType {
+	return gi.parent.GetStoreType()
+}
+
+// Implements KVStore.
+func (gi *gasKVStore) Get(key []byte) (value []byte) {
+	gi.gasMeter.ConsumeGas(ReadCostFlat, "GetFlat")
+	value = gi.parent.Get(key)
+	// TODO overflow-safe math?
```

Comment:
lets open an issue for it

Reply:
Ref https://github.com/cosmos/cosmos-sdk/issues/869